src/utils/dialog-validator.tsx

```diff
@@ -10,19 +10,15 @@ type ValidatorProps = {
   onChange: (isValid: boolean | string) => void;
   render: (hasError: boolean) => React.ReactElement<any>;
   isRequired: boolean;
+  isUniqName?: boolean;
```

Comment:
Rather than adding a new prop each time a new validation is required, consider providing an array of validators. Each validator can be for example a function that returns a string with error if value is invalid. This would lead to more reusable component and a common module of validation functions.